src/static/js/ours/client.js

```diff
@@ -31,13 +31,64 @@ CODALAB.api = {
         query = query || ''
         return CODALAB.api.request('GET', URLS.API + "competitions/" + query)
     },
-    create_competition: function(data) {
+    create_competition: function (data) {
         return CODALAB.api.request('POST', URLS.API + "competitions/", data)
     },
-    update_competition: function(data, pk) {
+    update_competition: function (data, pk) {
         return CODALAB.api.request('PATCH', URLS.API + "competitions/" + pk + "/", data)
     },
 
+    /*---------------------------------------------------------------------
+         Submissions
+    ---------------------------------------------------------------------*/
+    get_submissions: function (query, type) {
+        // return CODALAB.api.request('GET', URLS.API + `submissions/?q=${query || ''}&type=${type || ''}`)
+        return CODALAB.api.request('GET', URLS.API + `submissions/`)
+    },
+    delete_submission: function (id) {
+        return CODALAB.api.request('DELETE', URLS.API + "submissions/" + id + "/")
+    },
+    create_submission: function (form_data, progress_update_callback) {
+        // NOTE: this function takes a special "form_data" not like the normal
+        // dictionary other methods take
+
+
+        /*
+            Set variable CODALAB.IS_SERVER_LOCAL_STORAGE = true or false via context variable
+
+            Local storage:
+                * POST to server
+
+            Remote storage:
+                * POST to server
+                * Server returns SAS URL
+                * PUT to SAS URL
+                * POST to mark upload as done, so un-finished uploads can be pruned later
+
+        */
+        return $.ajax({
```

Comment:
can we abstract this nice function out? I think it's basically exactly the same for dataset/submission?